tests/test_aiokafka/test_aiokafka_consumer.py

```diff
@@ -91,10 +241,75 @@ async def test_subscribe(self):
 
         self.assertEqual(self.consumer.subscribed_topic, topics)
 
+    async def test_subscribe_pattern(self):
+        test_topic_2 = "test_topic_2"
+        self.kafka.create_partition(topic=self.test_topic, partitions=10)
+        self.kafka.create_partition(topic=test_topic_2, partitions=10)
+        self.kafka.create_partition(topic="other_topic", partitions=10)
+
+        self.consumer.subscribe(pattern=r"^test_.*")
+
+        topics = [self.test_topic, test_topic_2]
+        self.assertEqual(self.consumer.subscribed_topic, topics)
+
     async def test_subscribe_topic_not_exist(self):
-        topics = [self.test_topic]
+        self.create_topic()
+        topics = [self.test_topic, "missing-topic"]
         self.consumer.subscribe(topics=topics)
 
+        self.assertEqual(self.consumer.subscribed_topic, [self.test_topic])
+
+    async def test_lifecycle(self):
+        test_topic_2 = "test_topic_2"
+        self.kafka.create_partition(topic=self.test_topic, partitions=10)
+        self.kafka.create_partition(topic=test_topic_2, partitions=10)
+
+        topics = [self.test_topic, test_topic_2]
+        self.consumer.subscribe(topics=topics)
+
+        self.assertEqual(self.consumer.subscribed_topic, topics)
+
+        await self.consumer.start()
+
+        self.assertEqual(self.consumer.subscribed_topic, topics)
+
+        await self.produce_message()
+
+        messages = {
+            tp: self.summarise(msgs)
+            for tp, msgs in (await self.consumer.getmany()).items()
+        }
+        self.assertEqual(
+            {
+                TopicPartition(self.test_topic, partition=0): [
+                    ("test", "test"),
+                    ("test1", "test1"),
+                ],
+            },
+            messages,
+        )
+
+        await self.consumer.stop()
+
+        self.assertEqual(self.consumer.subscribed_topic, topics)
+
+        await self.consumer.start()
+        # TODO: work out what the expected behaviour is here -- should we start
+        # from a zero offset, or should that also persist?
```

Comment:
I don't have a lot of experience with Kafka -- certainly not enough to know what would be the expected default here. This also feels like something where it miiight be reasonable for the testing library to differ from a production library, though equally if it's something that one would expect to be able to configure in production then just ensuring it's configurable for the testing library might be the better option.

Unless we have an answer to hand, I suggest that we break this out to an issue rather than blocking this PR on resolving it here.